### PR TITLE
Egyptian raiders

### LINKED ISSUE #4845: citizens death by cramming
### Minecolonies version

Version: minecolonies-0.11.907-ALPHA-universal.jar
structurize-0.10.306-ALPHA.jar
forge-1.15.2-31.0.1

### Expected behavior

- 

### Actual behaviour

- 

### Steps to reproduce the problem

1. 
2. 
3. 
4. ... Profit

Hi Guys

Today I tried minecolonies-0.11.907-ALPHA-universal.jar, because in was waiting for a fix with the Issue #4701. That went fine, after starting I had 9 new Kid´s and could go on, growing up the colonie

Later on, I found several citizens are died, because of "cramming" ¿
so, what does that mean ? Did not have that before. What do you mean with cramming ?
What can I do ?

[02Mai2020 09:20:01.117] [Render thread/INFO] [net.minecraft.client.gui.NewChatGui/]: [CHAT]  Der Ritter Bella F. Smith starb bei 849 59 745 durch cramming!
[02Mai2020 09:50:42.486] [Render thread/INFO] [net.minecraft.client.gui.NewChatGui/]: [CHAT]  Der Ritter Lillian E. Culpepper starb bei 849 58 745 durch cramming!
[02Mai2020 09:51:40.193] [Render thread/INFO] [net.minecraft.client.gui.NewChatGui/]: [CHAT]  Der Healer Barnaby A. Barstaple starb bei 848 58 747 durch cramming!
[02Mai2020 11:25:57.749] [Render thread/INFO] [net.minecraft.client.gui.NewChatGui/]: [CHAT]  Der Brecher Julia F. Gooder starb bei 841 67 681 durch cramming!
[02Mai2020 12:51:33.203] [Render thread/INFO] [net.minecraft.client.gui.NewChatGui/]: [CHAT]  Der Ritter Benjamin S. Culpepper starb bei 849 58 746 durch cramming!
[02Mai2020 14:26:37.623] [Render thread/INFO] [net.minecraft.client.gui.NewChatGui/]: [CHAT]  Der Ritter Helen I. Ayde starb bei 848 59 745 durch cramming!

thx & regards

Inomutso

### Logs: Latest.log and crashlog at https://gist.github.com/

- latest.log:
- crashlog:

### LINKED ISSUE #4858: Raid crashes server
### Minecolonies version

Version: minecolonies-0.11.910-ALPHA-universal.jar

### Actual behaviour

- The server crashes sometimes, when a raid happens. Seems to be related to arraylist returning 0 as size.

### Logs: Latest.log and crashlog at https://gist.github.com/

- crashlog: [Pastebin](https://pastebin.com/VzhrhQkZ)

### PR DESCRIPTION
**Pull request by Raycoms**
Closes #4845
Closes #4858

# Changes proposed in this pull request:
- Fixes cramming issue of minecart
- Adds new raider type: Egyptian:
-> Spawn in the desert
-> Spawn also during the day if it rains
- Adds new item: Pharao Scepter
(Shoots fire arrows for free)
- Adds new item: Fire arrows

Review please

### CODE REVIEW

**Comment by LDTTeam-Buildsystem** [comment]
This PR has been automatically built and hosted by our build system, during this process there was a crash, error, or otherwise, please check on this at: https://teamcity.minecolonies.com/viewLog.html?buildId=38818&buildTypeId=LetSDevTogether_Minecolonies_PullRequests_2_DeployTestServer

File: src/main/java/com/minecolonies/coremod/entity/mobs/aitasks/EntityAIAttackArcher.java [hunk 1]
```diff
@@ -117,8 +123,15 @@ private void attack(final LivingEntity target)
...
             if (lastAttack <= 0 && entity.canEntityBeSeen(target))
             {
-                final ArrowEntity arrowEntity = EntityType.ARROW.create(target.world);
+                AbstractArrowEntity arrowEntity = EntityType.ARROW.create(target.world);
                 arrowEntity.setShooter(entity);
+
+                final ItemStack bow = entity.getHeldItem(Hand.MAIN_HAND);
+                if (bow.getItem() instanceof BowItem)
+                {
+                    arrowEntity = ((BowItem) bow.getItem()).customeArrow(arrowEntity);
```
Comment (Asherslab) [inline src/main/java/com/minecolonies/coremod/entity/mobs/aitasks/EntityAIAttackArcher.java:132]: type, `custom` please =D

File: src/main/java/com/minecolonies/coremod/items/ItemPharaoScepter.java [hunk 2]
```diff
@@ -0,0 +1,134 @@
...
+import net.minecraft.util.Hand;
+import net.minecraft.util.SoundCategory;
+import net.minecraft.util.SoundEvents;
+import net.minecraft.world.World;
+import org.jetbrains.annotations.NotNull;
+
+import java.util.function.Predicate;
+
+/**
+ * Class handling the Chief Sword item.
```
Comment (Asherslab) [inline src/main/java/com/minecolonies/coremod/items/ItemPharaoScepter.java]: javadoc

File: src/main/java/com/minecolonies/coremod/items/ItemPharaoScepter.java [hunk 3]
```diff
@@ -0,0 +1,134 @@
...
+ */
+public class ItemPharaoScepter extends BowItem
+{
+    /**
+     * Constructor method for the Chief Sword Item
+     *
+     * @param properties the properties.
+     */
+    public ItemPharaoScepter(final Properties properties)
+    {
```
Comment (Asherslab) [inline src/main/java/com/minecolonies/coremod/items/ItemPharaoScepter.java:35]: javadoc

File: src/main/resources/assets/minecolonies/lang/en_us.json [hunk 4]
```diff
@@ -789,7 +791,7 @@
 
   "block.minecolonies.blockpaperwall": "Paper wall",
 
-  "com.minecolonies.coremod.gui.workerhuts.buildprio": "Delivery Priority:",
+  "com.minecolonies.coremod.gui.workerhuts.buildprio": "Gathering Priority:",
```
Comment (Asherslab) [inline src/main/resources/assets/minecolonies/lang/en_us.json:794]: i suggest prefixing this with "Item", confusing otherwise

**Review by Asherslab — CHANGES_REQUESTED** [review]
few things

File: src/main/java/com/minecolonies/coremod/entity/mobs/aitasks/EntityAIAttackArcher.java (hunk 1 above)
Comment (Raycoms) [reply src/main/java/com/minecolonies/coremod/entity/mobs/aitasks/EntityAIAttackArcher.java:132]: this is the vanilla method name, this is not my personal choice =D

File: src/main/resources/assets/minecolonies/lang/en_us.json (hunk 4 above)
Comment (Raycoms) [reply src/main/resources/assets/minecolonies/lang/en_us.json:794]: too big for the GUI

File: src/main/java/com/minecolonies/coremod/entity/mobs/aitasks/EntityAIAttackArcher.java (hunk 1 above)
Comment (Asherslab) [reply src/main/java/com/minecolonies/coremod/entity/mobs/aitasks/EntityAIAttackArcher.java:132]: that's annoying, guess MCP has a typo then lol

**Review by Asherslab — DISMISSED** [review]

File: src/api/java/com/minecolonies/api/client/render/modeltype/EgyptianModel.java [hunk 5]
```diff
@@ -0,0 +1,20 @@
...
+
+import com.minecolonies.api.entity.mobs.egyptians.AbstractEntityEgyptian;
+import net.minecraft.client.renderer.entity.model.BipedModel;
+
+/**
+ * Egyptian model.
+ */
+public class EgyptianModel<T extends AbstractEntityEgyptian> extends BipedModel<AbstractEntityEgyptian>
+{
+    public EgyptianModel(final float size)
```
Comment (someaddons) [inline src/api/java/com/minecolonies/api/client/render/modeltype/EgyptianModel.java:15]: javadoc

File: src/api/java/com/minecolonies/api/client/render/modeltype/EgyptianModel.java [hunk 6]
```diff
@@ -0,0 +1,20 @@
...
+    {
+        super(size);
+    }
+
+    public EgyptianModel()
```
Comment (someaddons) [inline src/api/java/com/minecolonies/api/client/render/modeltype/EgyptianModel.java:23]: javadoc

File: src/api/java/com/minecolonies/api/entity/MinecoloniesMinecart.java [hunk 7]
```diff
@@ -282,6 +282,17 @@ public void applyEntityCollision(@NotNull final Entity entityIn)
         // Do nothing
     }
 
+    @Override
+    public void tick()
+    {
+        super.tick();
+
+        if (this.ticksExisted % 20 == 0 && getPassengers().isEmpty())
```
Comment (someaddons) [inline src/api/java/com/minecolonies/api/entity/MinecoloniesMinecart.java]: maybe == 19 or so? to make sure it doesnt despawn right when you spawn it

File: src/main/java/com/minecolonies/apiimp/initializer/EntityInitializer.java [hunk 8]
```diff
@@ -111,6 +118,35 @@ public static void setupEntities()
...
+                               .setRegistryName("mummy");
+
+        ModEntities.ARCHERMUMMY = EntityType.Builder.create(EntityArcherMummy::new, EntityClassification.MONSTER)
+                                     .setTrackingRange(ENTITY_TRACKING_RANGE)
+                                     .setUpdateInterval(ENTITY_UPDATE_FREQUENCY)
+                                     .size((float) CITIZEN_WIDTH, (float) CITIZEN_HEIGHT)
+                                     .build(Constants.MOD_ID + ":archermummy")
+                                     .setRegistryName("archermummy");
+
+        ModEntities.PHARAO = EntityType.Builder.create(EntityPharao::new, EntityClassification.MONSTER)
```
Comment (someaddons) [inline src/main/java/com/minecolonies/apiimp/initializer/EntityInitializer.java:136]: Did you try a different size for the boss? might look nice

File: src/main/java/com/minecolonies/coremod/client/model/raiders/ModelArcherMummy.java [hunk 9]
```diff
@@ -0,0 +1,143 @@
+package com.minecolonies.coremod.client.model.raiders;
+
+import com.minecolonies.api.client.render.modeltype.EgyptianModel;
+import com.minecolonies.api.entity.mobs.egyptians.AbstractEntityEgyptian;
+import net.minecraft.client.renderer.model.ModelRenderer;
+
+/**
+ * ModelArcherMummy - Either Mojang or a mod author
+ * Created using Tabula 7.0.0
```
Comment (someaddons) [inline src/main/java/com/minecolonies/coremod/client/model/raiders/ModelArcherMummy.java:9]: bit generic description? xD

File: src/main/java/com/minecolonies/coremod/client/model/raiders/ModelArcherMummy.java [hunk 10]
```diff
@@ -0,0 +1,143 @@
...
+        bipedRightArm.addChild(stripRightB);
+        bipedLeftLeg.addChild(legLeftLayer);
+
+        bipedHeadwear.showModel = false;
+    }
+
+    /**
+     * This is a helper function from Tabula to set the rotation of model parts
+     */
+    public void setRotateAngle(ModelRenderer modelRenderer, float x, float y, float z)
```
Comment (someaddons) [inline src/main/java/com/minecolonies/coremod/client/model/raiders/ModelArcherMummy.java:144]: params javadoc

File: src/main/java/com/minecolonies/coremod/client/model/raiders/ModelArcherMummy.java [hunk 11]
```diff
@@ -0,0 +1,143 @@
...
+import com.minecolonies.api.entity.mobs.egyptians.AbstractEntityEgyptian;
+import net.minecraft.client.renderer.model.ModelRenderer;
+
+/**
+ * ModelArcherMummy - Either Mojang or a mod author
+ * Created using Tabula 7.0.0
+ */
+public class ModelArcherMummy extends EgyptianModel<AbstractEntityEgyptian>
+{
+    public ModelArcherMummy()
```
Comment (someaddons) [inline src/main/java/com/minecolonies/coremod/client/model/raiders/ModelArcherMummy.java:16]: javadoc

File: src/main/java/com/minecolonies/coremod/client/model/raiders/ModelMummy.java [hunk 12]
```diff
@@ -0,0 +1,100 @@
+package com.minecolonies.coremod.client.model.raiders;
+
+import com.minecolonies.api.client.render.modeltype.EgyptianModel;
+import com.minecolonies.api.entity.mobs.egyptians.AbstractEntityEgyptian;
+import net.minecraft.client.renderer.model.ModelRenderer;
+
+/**
+ * ModelMummy - Either Mojang or a mod author
```
Comment (someaddons) [inline src/main/java/com/minecolonies/coremod/client/model/raiders/ModelMummy.java]: description again

File: src/main/java/com/minecolonies/coremod/client/model/raiders/ModelMummy.java [hunk 13]
```diff
@@ -0,0 +1,100 @@
...
+        bipedRightArm.addChild(stripRightA);
+        bipedRightLeg.addChild(legRightLayer);
+
+        bipedHeadwear.showModel = false;
+    }
+
+    /**
+     * This is a helper function from Tabula to set the rotation of model parts
+     */
+    public void setRotateAngle(ModelRenderer modelRenderer, float x, float y, float z)
```
Comment (someaddons) [inline src/main/java/com/minecolonies/coremod/client/model/raiders/ModelMummy.java:98]: javadoc

File: src/main/java/com/minecolonies/coremod/client/render/FireArrowRenderer.java [hunk 14]
```diff
@@ -0,0 +1,38 @@
...
+                                                     new ResourceLocation(Constants.MOD_ID, "textures/items/magicalarrows/magical_arrow6.png")
+                                                   };
+
+    public FireArrowRenderer(EntityRendererManager manager)
+    {
+        super(manager);
+    }
+
+    @NotNull
+    public ResourceLocation getEntityTexture(@NotNull final AbstractArrowEntity entity)
```
Comment (someaddons) [inline src/main/java/com/minecolonies/coremod/client/render/FireArrowRenderer.java:35]: javadoc

**Review by someaddons — DISMISSED** [review]

File: src/main/java/com/minecolonies/apiimp/initializer/EntityInitializer.java (hunk 8 above)
Comment (Raycoms) [reply src/main/java/com/minecolonies/apiimp/initializer/EntityInitializer.java:136]: No, didn't try yet =D

**Pull request merged**